### PR TITLE
[io] Migrate mutex, locks and cvs to C++14

### PR DESCRIPTION
**Pull request by SergioRAgostinho**
Let's see if this one compiles. It is a particularly challenging one because I'm not able to compile all these grabbers. I fear I might have forgotten some includes here and there. 🤞

### CODE REVIEW

File: io/include/pcl/io/impl/synchronized_queue.hpp [hunk 1]
```diff
@@ -38,10 +38,10 @@
 /* SynchronizedQueue Template, taken from
  * http://stackoverflow.com/questions/10139251/shared-queue-c
  */
+#pragma once
```
Comment (SergioRAgostinho) [inline io/include/pcl/io/impl/synchronized_queue.hpp]: I'm not really sure why I made this to the if guards. I can revert it later if desired.

**Comment by SergioRAgostinho** [comment]
A comment to discuss later, just noticed this in the CI log
```
-- The following subsystems will be built:
--   common
--   octree
--   io
--   kdtree
--   geometry
--   search
--   visualization
--   surface
--   sample_consensus
--   filters
--   2d
--   features
--   simulation
--   tracking
--   ml
--   segmentation
--   registration
--   recognition
--   outofcore
--   keypoints
--   stereo
--   examples
--   global_tests
--   tests_2d
--   tests_common
--   tests_features
--   tests_filters
--   tests_geometry
--   tests_io
--   tests_kdtree
--   tests_keypoints
--   tests_octree
--   tests_outofcore
--   tests_recognition
--   tests_registration
--   tests_search
--   tests_surface
--   tests_segmentation
--   tests_sample_consensus
--   tests_visualization
--   tools
--   apps
       building: 
       |_ in_hand_scanner
       |_ 3d_rec_framework
       |_ cloud_composer
       |_ point_cloud_editor
       |_ modeler
--   people
-- The following subsystems will not be built:
--   tests_people: Requires people.
```

Probably unrelated to this PR.

File: io/include/pcl/io/impl/synchronized_queue.hpp (hunk 1 above)
Comment (taketwo) [reply io/include/pcl/io/impl/synchronized_queue.hpp]: You probably realized that the current one is not correct (`H` vs `HPP`), leave alone the fact that we were supposed to switch to `#pragma once` anyway.

**Comment by taketwo** [comment]
Tutorials pipeline already failed, seems like an include is missing.

> Probably unrelated to this PR.

Looking at Azure test analytics this test wasn't executed a single time in 30 days, so it's been a while since it's off.

**Pull request merged**